src/test/ui/consts/zst_no_llvm_alloc.rs

```diff
@@ -0,0 +1,19 @@
+// run-pass
+
+#[repr(align(4))]
+struct Foo;
+
+static FOO: Foo = Foo;
+
+fn main() {
+    let x: &'static () = &();
+    assert_eq!(x as *const () as usize, 1);
+    let x: &'static Foo = &Foo;
+    assert_eq!(x as *const Foo as usize, 4);
```

Comment:
Does this constitute a *stable guarantee* that we will allocate ZSTs at certain addresses? I hope not, that seems quite arbitrary to me. After all we also don't guarantee where non-ZST are allocated.